Code/GeorgiaLibrarySystem-/Tests/UnitTest/MaterialTests.cs

```diff
@@ -0,0 +1,142 @@
+﻿using System;
+using System.Collections.Generic;
+using Core;
+using NUnit.Framework;
+using GTLService.DataAccess.Code;
+using GTLService.DataAccess.Database;
+using GTLService.DataManagement.Code;
+using Moq;
+
+namespace Tests.UnitTest
+{
+    public class MaterialTests
+    {
+        [Test]
+        //pass
+        [TestCase("", "", 10, null, "0")]
+        public void MaterialDmReadMaterialTest(string materialTitle, string author, int numOfRecords = 10, int isbn = 0, string jobStatus = "0")
+        {
+            //Arrange
+            var materialDa_Code_Mock = new Mock<MaterialDa_Code>(null);
+            var libraryDa_Code_Mock = new LibraryDa_Code(null);
+            var personDa_Code_Mock = new PersonDa_Code(null);
+            var objects = MaterialsSetUp();
+
+            materialDa_Code_Mock.Setup(x => x.ReadMaterials(It.IsAny<int>(), It.IsAny<string>(), It.IsAny<string>(), It.IsAny<int>()))
+                .Returns(objects.Item2);
+            materialDa_Code_Mock.Setup(x => x.ReadCopies(It.IsAny<int>(), It.IsAny<string>()))
+                .Returns(objects.Item1);
+
+            var materialDm = new MaterialDm_Code(materialDa_Code_Mock.Object, libraryDa_Code_Mock, personDa_Code_Mock);
+
+            //Act
+            var result = materialDm.ReadMaterials(materialTitle, author, numOfRecords, isbn, jobStatus);
+
+            //Assert
+            Assert.IsTrue(result.Count == 2);
+            Assert.IsTrue(result[0].Available_Copies == 2);
+            Assert.IsTrue(result[1].Available_Copies == 1);
```

Comment:
didnt teachers say that we shouldnt have more then one assert in test?

Reply:
teachers said its bad to test for more than 1 thing at a time, nothing about asserts

Reply:
but arent you testing more then one thing at the time? i mean you check for count and the available copies